src/test/ui/command/command-create-pidfd.rs

```diff
@@ -2,10 +2,45 @@
 // only-linux - pidfds are a linux-specific concept
 
 #![feature(linux_pidfd)]
-use std::os::linux::process::{CommandExt, ChildExt};
+#![feature(rustc_private)]
+
+extern crate libc;
+
+use std::ffi::CStr;
+use std::io::Error;
+use std::mem::MaybeUninit;
+use std::os::linux::process::{ChildExt, CommandExt};
 use std::process::Command;
 
+/// Returns the version of the Linux kernel as (RELEASE, MAJOR, MINOR)
+fn kernel_version() -> (u32, u32, u32) {
+    let mut buf = MaybeUninit::zeroed();
+    let release = match unsafe { libc::uname(buf.as_mut_ptr()) } {
+        -1 => Err(Error::last_os_error()),
+        _ => Ok(unsafe { buf.assume_init() }),
+    }
+    .expect("failed to retrieve kernel version")
+    .release;
+    let version = unsafe { CStr::from_ptr(release[..].as_ptr()) }
+        .to_str()
+        .unwrap()
+        .split('-')
+        .next()
+        .unwrap();
+    let mut version_iter = version.split('.').map(|s| s.parse().unwrap());
+    (
+        version_iter.next().unwrap(),
+        version_iter.next().unwrap(),
+        version_iter.next().unwrap(),
+    )
+}
+
 fn main() {
+    // pidfds require the clone3 syscall which was only added in Linux 5.3
+    if kernel_version() < (5, 3, 0) {
```

Comment:
I think it would be better to try calling `clone3` directly, and test for `ENOSYS`. To prevent the syscall from doing anything when it exists, we could pass an invalid flag combination to force it return an error. However, that doesn't need to block this PR.

Reply:
Okay, I changed it.